sympy/stats/rv_interface.py

```diff
@@ -483,6 +483,41 @@ def coskewness(X, Y, Z, condition=None, **kwargs):
          * std(Z, condition, **kwargs)
     return num/den
 
+def self_information(p):
+    """
+    Calculates the self-information of a particular outcome of an event.
+    It represents the degree of surprise associated with a certain outcome.
+    Mathematically it is defined as
+
+    .. math::
+        selfinformation(p)=-\log_2 (p)
+
+    Parameters
+    ==========
+
+    p : Probability of an event
```

Comment:
I think, instead of probability we should accept the same arguments as `probability` a.k.a `P` and return self information. In addition, support for any base `b` should be there rather than a fixed `2`. 
P.S. What's the point if one has to call, `self_information` rather than just doing `-log(p, b)`.

Reply:
Thanks a lot, I had this doubt too, the function looked way too simple. As for the base 2, I thought that was the convention, every source had suggested that. I'll change it regardless.